django/contrib/auth/tokens.py

```diff
@@ -82,12 +79,12 @@ def _make_hash_value(self, user, timestamp):
         login_timestamp = '' if user.last_login is None else user.last_login.replace(microsecond=0, tzinfo=None)
         return str(user.pk) + user.password + str(login_timestamp) + str(timestamp)
 
-    def _num_days(self, dt):
-        return (dt - date(2001, 1, 1)).days
+    def _num_seconds(self, dt):
+        return int((dt - datetime(2001, 1, 1)).total_seconds())
 
-    def _today(self):
+    def _now(self):
         # Used for mocking in tests
-        return date.today()
+        return datetime.now()
```

Comment:
I think we can remove microseconds `datetime.now().replace(microsecond=0)` and after that we can also remove an `int()` call in `_num_seconds()` :thinking:

Reply:
If we remove `int()` call in `_num_seconds()`, it returns a number in `float` and cause problem in 
https://github.com/django/django/blob/bae05bcf68710cb6dafa51325c3ec83ddda83c39/django/utils/http.py#L229

`TypeError: string indices must be integers`

Reply:
I missed that, thanks.